### PR TITLE
Fix no diagnostics without WorkspaceFolders support

### PR DESCRIPTION
**Pull request by a5ob7r**
Document diagnostics by shellcheck doesn't work without WorkspaceFolders support clients, because no response from the client about WorkspaceFolders and it blocks subsequent steps. To resolve this, send the request only if the client supports WorkspaceFolders.

### CODE REVIEW

File: server/src/server.ts [hunk 1]
```diff
@@ -53,17 +53,20 @@ export default class BashServer {
 
   private documents: LSP.TextDocuments<TextDocument> = new LSP.TextDocuments(TextDocument)
   private connection: LSP.Connection
+  private clientCaps: LSP.ClientCapabilities
```
Comment (skovhus) [inline server/src/server.ts:56]: Nitpick: but let us spell this out. clientCapabilities.

**Review by skovhus — COMMENTED** [review]
Thanks for fixing this. 

I’m curious: Which clients doesn’t support this?

File: server/src/server.ts (hunk 1 above)
Comment (a5ob7r) [reply server/src/server.ts:56]: I got it.

**Comment by a5ob7r** [comment]
> I’m curious: Which clients doesn’t support this?

It's [vim-lsp](https://github.com/prabirshrestha/vim-lsp). Actually it supports WorkspaceFolders, but that feature is experimental and disabled by default now.
https://github.com/prabirshrestha/vim-lsp/blob/da80b018311b811eceac86fd0a59c521ffa9d2ab/doc/vim-lsp.txt#L2033

**Review by skovhus — APPROVED** [review]

**Comment by skovhus** [comment]
Released as 3.0.5. Does it work for you @a5ob7r?

**Comment by a5ob7r** [comment]
@skovhus Yes, it works fine. Thank you!

**Pull request merged**